symengine/lib/symengine_wrapper.pyx

```diff
@@ -467,49 +482,24 @@ cdef class Basic(object):
         cdef Basic s = sympify(x)
         return c2py(symengine.diff(self.thisptr, s.thisptr))
 
-    def subs_dict(Basic self not None, subs_dict):
-        cdef _DictBasic D
-        if isinstance(subs_dict, DictBasic):
-          D = subs_dict
-          return c2py(symengine.ssubs(self.thisptr, D.c))
-        cdef symengine.map_basic_basic d
-        cdef Basic K, V
-        for k in subs_dict:
-            K = sympify(k)
-            V = sympify(subs_dict[k])
-            d[K.thisptr] = V.thisptr
-        return c2py(symengine.ssubs(self.thisptr, d))
+    #TODO: deprecate this
+    def subs_dict(Basic self not None, *args):
```

Comment:
Just add the deprecation warnings in this PR, that way we don't need to revisit in the future.